app/models/dataset_record/darwin_core/occurrence.rb

```diff
@@ -298,6 +298,16 @@ def clear_empty_sub_hashes(hsh)
     hsh
   end
 
+  # Remove the namespace short name and delimiter from start of string.
+  #
+  # If the namespace has a verbatim_short_name, that is removed instead of the short_name.
+  # The delimiter is only removed if the short_name was found in the identifier.
+  # @param [String] identifier_str
+  # @param [Namespace] namespace
+  def delete_namespace_prefix(identifier_str, namespace)
```

Comment:
Please rename to `delete_namespace_prefix!` to indicate this method alters supplied `identified_str` rather than returning a modified copy.